### ISSUE TITLE
[Issue] When I run azd init in the wrong place, it by default dumps a lot of files there before I realize its the wrong place.

### ISSUE THREAD

**Issue by Petermarcu** [Azure/azure-dev#1638]
We should consider warning before creating a bunch of files that the folder we're pointing to isn't empty.

**Comment by weikanglim** [comment]
@savannahostrowski Now that we're initializing a `git` repository, we might need to revisit the behavior. If the user runs `azd init` unintentionally in a root folder, we end up creating a `git` repository that is: hard to detect, and hard to cleanup, causes issues with all directories that now will be rooted to the parent `git`  directory.

I think `azd init` in a non-empty folder should immediately fail. I think we should fix the error message to contain useful follow-ups, and support flags to switch between the modes:

ERROR: unable to initialize in directory '<folder>', directory is non-empty.
- Did you mean to initialize in a new directory? Run `azd init --output <directory>`
- Did you mean to overwrite existing files? Run `azd init --force`

**Comment by savannahostrowski** [comment]
Yeah - I agree. That behaviour makes sense to me. I think for the first "did you mean.." we can also tell them to `cd` into that dir first.

As an aside, I swear that there's a dupe of this issue floating around but I've been unable to find it.

**Comment by savannahostrowski** [comment]
If we detect that the user is not in an empty directory, we should print out something like

```
 "Error: <somedirectory> is not empty. Either change into an empty directory before running this command, or rerun this command with --force"
```

**Comment by weikanglim** [comment]
@savannahostrowski
We definitely want to move the "initialize from a template to update my existing project" to an opt-in flag. That way, we can now break on non-empty directory for normal flows.

## Proposal

User specifies a flag to opt-in to update existing project behavior.

Possible flag names:

- `azd init --update --template <t>` 
- `azd init --refresh --template <t>` 

I lean towards `--update`.

--update: Updates existing application (project?) files in the current directory from an existing template. 

In this mode, we can at least validate that `azure.yaml` exists.

**Comment by weikanglim** [comment]
Aside from supporting users that want to update their project based on an existing template, we might also want to consider how `azd init` would play nicely with azdevifying an existing application codebase. I think `azd init` should still work when you run it inside an existing codebase.

We can either support this via a flag, or a prompt. I'm leaning towards a prompt that looks like:

When user is running it in a non-empty directory without `.git`:

> The current directory is non-empty. Running `azd init` will places files and initialize a git repository in '~/repos/todo`. Continue?

When user is running in non-empty directory with git:

> The current directory is non-empty. Running `azd init` will place files in '~/repos/todo`. Continue?

**Comment by weikanglim** [comment]
There's also a question of a gesture (this may be a separate issue) to create the project in a new directory. Maybe taking inspiration from [cargo](https://users.rust-lang.org/t/cargo-new-vs-cargo-init/40794), we could have:

`azd init` vs. `azd new`

We can wait for user feedback on this.
